src/playlist-controller.js

```diff
@@ -2404,21 +2406,39 @@ export class PlaylistController extends videojs.EventTarget {
       }
       keyIdSet.forEach((key) => {
         const USABLE = 'usable';
-        const NON_USABLE = 'non-usable';
         const hasUsableKeyStatus = this.keyStatusMap_.has(key) && this.keyStatusMap_.get(key) === USABLE;
         const nonUsableExclusion = playlist.lastExcludeReason_ === NON_USABLE && playlist.excludeUntil === Infinity;
 
         if (!hasUsableKeyStatus) {
-          playlist.excludeUntil = Infinity;
-          playlist.lastExcludeReason_ = NON_USABLE;
-          this.logger_(`excluding playlist ${playlist.id} because the key ID ${key} doesn't exist in the keyStatusMap or is not ${USABLE}`);
+          // Only exclude playlists that haven't already been excluded as non-usable.
+          if (!nonUsableExclusion && playlist.lastExcludeReason_ !== NON_USABLE) {
+            playlist.excludeUntil = Infinity;
+            playlist.lastExcludeReason_ = NON_USABLE;
+            this.logger_(`excluding playlist ${playlist.id} because the key ID ${key} doesn't exist in the keyStatusMap or is not ${USABLE}`);
+          }
+          // count all nonUsableKeyStatus
+          nonUsableKeyStatusCount++;
         } else if (hasUsableKeyStatus && nonUsableExclusion) {
           delete playlist.excludeUntil;
           delete playlist.lastExcludeReason_;
           this.logger_(`enabling playlist ${playlist.id} because key ID ${key} is ${USABLE}`);
         }
       });
     });
+
+    // If for whatever reason every playlist has a non usable key status. Lets try re-including the SD renditions as a failsafe.
+    if (nonUsableKeyStatusCount >= this.mainPlaylistLoader_.main.playlists.length) {
+      this.mainPlaylistLoader_.main.playlists.forEach((playlist) => {
+        const isNonHD = playlist && playlist.attributes && playlist.attributes.RESOLUTION && playlist.attributes.RESOLUTION.height < 720;
+        const excludedForNonUsableKey = playlist.excludeUntil === Infinity && playlist.lastExcludeReason_ === NON_USABLE;
```

Comment:
I think there's actually a scenario where a playlist for another reason, perhaps an incompatible codec, will have an `excludeUntil = Infinity` but not  `lastExcludeReason_ = NON_USABLE`. It might increment the `nonUsableKeyStatusCount` still if it doesn't have `usable` key in from the KeyStatusMap, which in turn could get us to this logic, however I don't think we would want to re-include that playlist if it has an incompatible codec for the browser.

Reply:
Good point, I think I was misunderstanding and assuming all of these would have the same exclusion reason.